src/odb/include/odb/geom.h

```diff
@@ -339,6 +343,10 @@ class Polygon
   int dx() const { return getRect().dx(); }
   int dy() const { return getRect().dy(); }
 
+  // returns a corrected Polygon with a closed form and counter-clockwise points
+  Polygon correct() const;
```

Comment:
Is there a reason to not make this automatic in the ctor / setPoints ?  I'm not sure why we would want an 'incorrect' one.

Reply:
I agree, but that would mean that dbPolygon would need to track the "original" points vs the correct points to allow the lefout to reflect back the points from the lefin (or alternatively, accept that the LEFout will always be a correct polygon). Any thoughts?

Reply:
So it is more of "make boost friendly" than "correct".  In that case perhaps it should be private as it seems to be specific to the bloat case.

Reply:
yes and no, to properly do `boost::geometry::intersect` the polygon has to be correct. Without it you can't rely on it. Personally, I think the `Polygon` should correct the points to forms a complete polygon otherwise all code that uses a polygon needs to first get the "correct" variant of it (so it could be private and handled in `setPoints`).
I use this function for that very reason in the RDL code, so it can't be private as long as the polygon may be incomplete.